File: 560. Subarray Sum Equals K.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        '''
        https://leetcode.com/articles/subarray-sum-equals-k/
        Also approach 2,3 are also good but will TLE on leetcode
        '''
        sum = 0
        count = 0
        dic = {}
        dic[0] = 1
        for num in nums:
            sum += num
            if sum-k in dic:
                count += dic[sum-k]
            dic[sum] = dic.get(sum, 0) + 1
        return count

File: 560. Subarray Sum Equals K.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int count=0, sum=0;
        HashMap<Integer, Integer> map = new HashMap<>();
        map.put(0,1);
        for (int i : nums) {
            sum += i;
            count += map.getOrDefault(sum-k, 0);
            map.put(sum, map.getOrDefault(sum, 0)+1);
        }
        return count;
    }
}